controllers/appwrapper_controller.go

```diff
@@ -223,28 +237,27 @@ func onAdd(obj interface{}) {
 	aw, ok := obj.(*arbv1.AppWrapper)
 	if ok {
 		klog.Infof("Found Appwrapper named %s that has status %v", aw.Name, aw.Status.State)
-		if aw.Status.State == arbv1.AppWrapperStateEnqueued || aw.Status.State == "" {
+		if aw.Status.State == arbv1.AppWrapperStateEnqueued || aw.Status.State == "" && aw.Labels["orderedinstance"] != "" {
 			//scaledAppwrapper = append(scaledAppwrapper, aw.Name)
 			demandPerInstanceType := discoverInstanceTypes(aw)
 			//TODO: simplify the looping
-			if useMachineSets {
-				if canScaleMachineset(demandPerInstanceType) {
-					scaleUp(aw, demandPerInstanceType)
+			if demandPerInstanceType != nil {
+				if useMachineSets {
+					if canScaleMachineset(demandPerInstanceType) {
+						scaleUp(aw, demandPerInstanceType)
+					} else {
+						klog.Infof("Cannot scale up replicas max replicas allowed is %v", maxScaleNodesAllowed)
+					}
 				} else {
-					klog.Infof("Cannot scale up replicas max replicas allowed is %v", maxScaleNodesAllowed)
-				}
-			} else {
-				if canScaleMachinepool(demandPerInstanceType) {
-					scaleUp(aw, demandPerInstanceType)
-				} else {
-					klog.Infof("Cannot scale up replicas max replicas allowed is %v", maxScaleNodesAllowed)
+					if canScaleMachinepool(demandPerInstanceType) {
+						scaleUp(aw, demandPerInstanceType)
+					} else {
+						klog.Infof("Cannot scale up replicas max replicas allowed is %v", maxScaleNodesAllowed)
+					}
 				}
 			}
```

Comment:
I noticed there was a TODO for simplifying this loop. This reads a bit better:
```
if demandPerInstanceType != nil {
	if (useMachineSets && canScaleMachineset(demandPerInstanceType)) || (!useMachineSets && canScaleMachinepool(demandPerInstanceType)) {
		scaleUp(aw, demandPerInstanceType)
	} else {
		klog.Infof("Cannot scale up replicas. The maximum allowed replicas is %v", maxScaleNodesAllowed)
	}
}
```

Reply:
Thanks @ChristianZaccaria,
I'm working on a follow up pr with refactors, which will include this :)

Reply:
That sounds great, well done on this PR Eoin, looks fantastic :+1: